### PR TITLE
refactor(test): switch sync_point by feature flag

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

To make using sync_point easier in unit tests.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/cmd_all/Cargo.toml [hunk 1]
```diff
@@ -5,6 +5,7 @@ edition = "2021"
 
 [features]
 static-link = ["workspace-static-link"]
+sync_point = ["sync-point/sync_point"]
```
Comment (wangrunji0408) [inline src/cmd_all/Cargo.toml:8]: Don't have to add a feature here. A top-level `--feature sync-point/sync_point` is enough.

File: src/meta/Cargo.toml [hunk 2]
```diff
@@ -74,3 +75,4 @@ tempfile = "3"
 [features]
 test = []
 failpoints = ["fail/failpoints"]
+sync_point = ["sync-point/sync_point"]
```
Comment (wangrunji0408) [inline src/meta/Cargo.toml:78]: same as above

**Review by wangrunji0408 — APPROVED** [review]
LGTM!

File: src/cmd_all/Cargo.toml (hunk 1 above)
Comment (zwang28) [reply src/cmd_all/Cargo.toml:8]: Right. Thanks.

**Review by soundOfDestiny — APPROVED** [review]
lgtm

**Comment by zwang28** [comment]
@mergifyio requeue

**Pull request merged**